### PR TITLE
feat: add IllegalResume error

### PR DESCRIPTION
**Pull request by mlutze**
related to #3543, depends on #3650.

### CODE REVIEW

**Comment by magnus-madsen** [comment]
Stupid question: What if `resume` was a special construct that we put into the local name context and then treat specially?

**Comment by mlutze** [comment]
> Stupid question: What if `resume` was a special construct that we put into the local name context and then treat specially?

So essentially what I am doing here, but instead in the `Namer` and attached to the environment?

File: main/src/ca/uwaterloo/flix/language/errors/WeederError.scala [hunk 1]
```diff
@@ -747,4 +747,29 @@ object WeederError {
...
+         |>> Unexpected use of 'resume'.
+         |
+         |${code(loc, "unexpected use of 'resume'")}
+         |
+         |""".stripMargin
+    }
+
+    def explain(formatter: Formatter): Option[String] = Some({
+      import formatter._
+      s"${underline("Tip:")} The 'resume' expression may only be used in effect handlers."
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/errors/WeederError.scala:771]: I think we should include the statement "resume must occur in an effect handler" in the summary and message itself, and then leave the explain = None.

File: main/src/ca/uwaterloo/flix/language/phase/Weeder.scala [hunk 2]
```diff
@@ -2879,4 +2888,10 @@ object Weeder {
     region.getOrElse(exp)
   }
 
+  private sealed trait SyntacticEnv
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Weeder.scala:2891]: Add some comments.

File: main/test/ca/uwaterloo/flix/language/phase/TestWeeder.scala [hunk 3]
```diff
@@ -660,4 +660,13 @@ class TestWeeder extends FunSuite with TestUtils {
     val result = compile(input, Options.TestWithLibNix)
     expectError[WeederError.IllegalEffectTypeParams](result)
   }
+
+  test("IllegalResume.01") {
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/language/phase/TestWeeder.scala:664]: Add a few more test cases.

**Review by magnus-madsen — COMMENTED** [review]
I added some comments. I will merge it now anyway, but I should you add these comments to a ticket.

**Pull request merged**